### ISSUE TITLE
cannot modify PoolVector2Array in child node from parent node

### ISSUE THREAD

**Issue by Triangle345** [godotengine/godot#17896]
**Godot version:**
3.0.2

** Description **
Basically I have a parent Node2D and a child Node2D. The child Node2D contains a PoolVector2Array which i cannot modify/access from the parent.  Here is an example:

Parent Node
```
extends Node2D


onready var my_node = get_node("MyNode")

func _ready():

    my_node.myvectors.append(Vector2(100,100))
    print ("my vectors size: ", my_node.myvectors.size()) # for some reason size is zero its like i cant modify the array??

func _process(delta):
    pass

```

Child Node:
```


extends Node2D

onready var myvectors = PoolVector2Array()
func _ready():

    pass

```

More information can be found on the reddit page below:

https://www.reddit.com/r/godot/comments/88lj9e/cannot_modify_array_from_child_node_unless_using/

**Comment by ywaby** [comment]
api doc
```
An Array specifically designed to hold Vector2. Note that this type is passed by value and not by reference.
```

**Comment by reduz** [comment]
Yes this is not possible, but its going to be modified so it is soon, it is in the list of compat breakage for 3.2/4.0

**Comment by KoBeWi** [comment]
I noticed that same thing happens when you try to add to `points` in Line2D. Interestingly, if you make a custom variable called `points2` of PoolVector2Array it will work.

This is still not fixed in f48aeeeee

**Comment by RichardEllicott** [comment]
i also have this problem it is annoying a little, but i think it might be since the way these arrays are stored

i find i have to copy the array out, i don't know if it copies the whole array of a reference, i do this when editing a Line2D points:

    var points = drawing_line_2d.points
    points.append(mouse_pos)
    drawing_line_2d.points = points`

so at least there is an easy workaround

**Comment by Zylann** [comment]
Same as https://github.com/godotengine/godot/issues/15265 ?

**Comment by KoBeWi** [comment]
Duplicate of #15265
Fixed in 4.0
